AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Support/Utils/CustomHttpEngine.swift

```diff
@@ -0,0 +1,16 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+@_spi(InternalCustomHttpEngine) @_spi(InternalAmplifyPluginExtension) import AWSPluginsCore
+import ClientRuntime
+import Foundation
+
+protocol CustomHttpEngine: HttpClientEngine {
+    func setHttpClientEngine(_ httpClientEngine: HttpClientEngine)
```

Comment:
This setter makes me think that this make the notion of a proxy a bit more solid. What do you think about this?

```
protocol HttpClientEngineProxy: HttpClientEngine {
    var target: HttpClientEngine?
}
```